proto/rill/runtime/v1/resources.proto

```diff
@@ -418,3 +419,29 @@ message ExecutionError {
 message CharLocation {
   uint32 line = 1;
 }
+
+message API {
+  APISpec spec = 1;
+  APIState state = 2;
+}
+
+// API provide SQL based access to data stored in Rill
+message APISpec {
+  // sql is free-form sql to query against any source/model stored in Rill
+  string sql = 1;
+  // metrics is used for querying against a metrics view
+  MetricSQL metrics = 2;
+}
+
+// MetricSQL is used for querying against a metrics view
+// If sql is set, it is parsed and executed as metrics SQL
+// Alternatively, you can also pass parameters similar to MetricsViewAggregation
+message MetricSQL {
+  string sql = 1;
+  string metrics_view = 2;
+  repeated string measures = 3;
+  string where = 4;
+  string limit = 5;
+}
```

Comment:
I think we can avoid putting the specific resolver properties here – it should be sufficient to parse them in the resolver implementation. With reference to the `APIResolverOptions` proposal in the design doc, I would consider something like this:
```proto3
message APISpec {
  string resolver = 1;
  google.protobuf.Struct resolver_properties = 2;
}
```

Reply:
Makes sense. 
But a user would not be expected to set resolver for custom APIs right ? We would need to infer it basis the properties. For built-in APIs a resolver would be required.